src/libraries/System.Net.Security/src/System/Net/CertificateValidationPal.Windows.cs

```diff
@@ -48,7 +48,11 @@ internal static partial class CertificateValidationPal
             SafeFreeCertContext? remoteContext = null;
             try
             {
-                remoteContext = SSPIWrapper.QueryContextAttributes_SECPKG_ATTR_REMOTE_CERT_CONTEXT(GlobalSSPI.SSPISecureChannel, securityContext);
+                // SECPKG_ATTR_REMOTE_CERT_CHAIN can be used even before the TLS handshake completes, which is necessary
+                // in order to supply the certificate to the client cert selection callback. However, it is not available on
+                // windows 7, so use the SECPKG_ATTR_REMOTE_CERT_CONTEXT as a fallback option.
+                remoteContext = SSPIWrapper.QueryContextAttributes_SECPKG_ATTR_REMOTE_CERT_CHAIN(GlobalSSPI.SSPISecureChannel, securityContext)
```

Comment:
Is the never called on server? It would be pretty normal that the remote certificate is null but it would not make sense to try again IMHO. We can probably create static property to show if the call is available - similar to what were do for TLS 1.3.